coding/trie.hpp

```diff
@@ -79,4 +83,19 @@ void ForEachRef(Iterator<TValue> const & iter, F & f, TString const & s)
   }
 }
 
-}  // namespace Trie
+template <typename TValueList, typename TF, typename TString>
+void ForEachRefWithValues(Iterator<TValueList> const & iter, TF && f, TString const & s)
```

Comment:
Agree, we need only one version (ForEachRefWithValues). ForEachRef() looks very strange, as it calls f() with the same argument s m_valueList.size() times.